interfaces/cython/cantera/thermo.pyx

```diff
@@ -632,7 +632,30 @@ cdef class ThermoPhase(_SolutionBase):
         stoichAirFuelRatio = - (Of - 2*Cf - 2*Sf - Hf/2.0) / (Oo - 2*Co - 2*So - Ho/2.0)
         Xr = phi * Xf + stoichAirFuelRatio * Xo
         self.TPX = None, None, Xr
+        
+    def get_equivalence_ratio(self):
+        """Considers the oxidation of C to CO2, H to H2O and S to SO2.
+        Other elements are assumed not to participate in oxidation
+        (i.e. N ends up as N2).
+        O2 is considered as the only oxidizer (i.e. NO is considered a "fuel")
+        """
+        alpha = 0
+        for s in self.species():
+            if s.composition == {'O': 2}:
+                mol_O2 = self[s.name].X[0]
+            else:
+                nC = s.composition.get('C', 0)
+                nH = s.composition.get('H', 0)
+                nO = s.composition.get('O', 0)
+                nS = s.composition.get('S', 0)
+
+                alpha += (nC + nH/4 + nS - nO/2) * self[s.name].X[0]
 
+        if mol_O2 == 0:
```

Comment:
Hmm, sorry just noticed two more things here. One is that there may be cases where `mol_O2` is very close to zero, but not identically (`1E-15` or something), and this comparison would result in an equivalence ratio that is huge. I'm not totally sure what the best way to handle that situation is (`if mol_O2 < some_small_value`, or something else, maybe @speth has an idea here), but I don't think we should be doing floating point equality comparisons.

Second, in the case where there is no species with the composition `{'O': 2}` in the mechanism, this will throw a `NameError`, which should be handled somehow. Again, not sure of the best way to handle this...

Reply:
I think that if `mol_O2` is very close to zero, it's okay that this would result in a huge equivalence ratio. With `mol_O2 == 1e-15` , then it would be accurate to return a huge equivalence ratio. I only put the case with `mol_O2 == 0` to avoid a division by zero error.

If the mechanism doesn't have a species with the composition `{'O': 2}`, then that means there is no molecular oxygen, so the equivalence ratio should be `inf`, right? I can add that exception handling if you agree.